server/app/uk/gov/ons/addressIndex/server/controllers/AddressController.scala

```diff
@@ -38,19 +44,45 @@ class AddressController @Inject()(
     * @param format requested format of the query (paf/nag)
     * @return Json response with addresses information
     */
-  def addressQuery(input: String, format: String): Action[AnyContent] = Action async { implicit req =>
-    logger info s"#addressQuery:\ninput $input , format: $format"
-    input.toOption map { actualInput =>
-      val tokens = parser tag actualInput
-      logger info s"#addressQuery parsed:\n${tokens.map(t => s"value: ${t.value} , label:${t.label}").mkString("\n")}"
-      formatQuery[AddressBySearchResponseContainer, Seq[CrfTokenResult]](
-        formatStr = format,
-        inputForPafFn = AddressQueryInput(tokens),
-        pafFn = pafSearch,
-        inputForNagFn = AddressQueryInput(tokens),
-        nagFn = nagSearch
-      ) getOrElse futureJsonBadRequest(UnsupportedFormat)
-    } getOrElse futureJsonBadRequest(EmptySearch)
+  def addressQuery(input: String, format: String, offset: Option[String] = None, limit: Option[String] = None): Action[AnyContent] = Action async { implicit req =>
```

Comment:
This is to tidily deal with the case where the query string contains non-numeric values. For example limit=fifty. The  controller attempts to convert "fifty" to an Int and if this fails a bad request with a message string indicating that limit param must be numeric returned. If limit and offset were Option[Int]s I'm not sure how this scenario would  pan out.